src/main/java/uk/gov/hmcts/reform/hmc/validator/ListingReasonCodeEnumPatternValidator.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.reform.hmc.validator;
+
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class ListingReasonCodeEnumPatternValidator implements ConstraintValidator<ListingReasonCodeEnumPattern,String> {
+    private List<String> acceptedValues;
+
+    @Override
+    public void initialize(final ListingReasonCodeEnumPattern annotation) {
+        acceptedValues = Stream.of(annotation.enumClass().getEnumConstants())
+            .map(Enum::name)
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        if (value == null) {
+            return true;
```

Comment:
Using EnumPatternValidator produces a ConstraintValidator error when listingAutoChangeReasonCode is null.
Created this validator to change the return to true when listingAutoChangeReasonCode is null